### PR TITLE
Support for BATCHSIZE, MINBATCHSIZE, INPUTS and OUTPUTS on AI.MODELGET

### LINKED ISSUE #5: Support for BATCHSIZE, MINBATCHSIZE, INPUTS and OUTPUTS on AI.MODELGET
further reference: https://oss.redislabs.com/redisai/commands/#aimodelget

### PR DESCRIPTION
**Pull request by filipecosta90**
fixes #5

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -86,14 +88,36 @@ export class Model {
...
+  private _batchsize: number;
+  private _minbatchsize: number;
+
   static NewModelFromModelGetReply(reply: any[]) {
     let backend = null;
     let device = null;
     let tag = null;
     let blob = null;
+    let batchsize: null | number = null;
+    let minbatchsize: null | number = null;
```
Comment (gkorland) [inline src/model.ts:114]: why not set the default to 0?

File: src/model.ts [hunk 2]
```diff
@@ -126,10 +167,43 @@ export class Model {
...
+    const model = new Model(backend, device, inputs, outputs, blob);
     if (tag !== null) {
       model.tag = tag;
     }
+    if (batchsize !== null) {
+      model.batchsize = batchsize;
+    }
+    if (minbatchsize !== null) {
+      model.minbatchsize = minbatchsize;
+    }
```
Comment (gkorland) [inline src/model.ts:179]: why not part of constructor?

Comment (filipecosta90) [reply src/model.ts:179]: you're right, but will add it as an optional parameter to it, given that most people will not use them at least on a first usage.

File: src/model.ts (hunk 1 above)
Comment (filipecosta90) [reply src/model.ts]: agree. solved on the new commit

File: src/model.ts (hunk 2 above)
Comment (filipecosta90) [reply src/model.ts]: solved on the new commit

File: src/model.ts [hunk 3]
```diff
@@ -106,9 +140,26 @@ export class Model {
...
+          batchsize = parseInt(obj.toString(), 10);
+          break;
+        case 'minbatchsize':
+          minbatchsize = parseInt(obj.toString(), 10);
+          break;
+        case 'inputs':
+          // tslint:disable-next-line:prefer-for-of
+          for (let j = 0; j < obj.length; j++) {
+            inputs.push(obj[j].toString());
+          }
```
Comment (gkorland) [inline src/model.ts:155]: obj.forEach(input =>  inputs.push(input.toString()))

File: src/model.ts [hunk 4]
```diff
@@ -106,9 +140,26 @@ export class Model {
...
+          break;
+        case 'outputs':
+          // tslint:disable-next-line:prefer-for-of
+          for (let j = 0; j < obj.length; j++) {
+            outputs.push(obj[j].toString());
+          }
```
Comment (gkorland) [inline src/model.ts:161]: obj.forEach(output =>  inputs.push(output.toString()))

File: src/model.ts (hunk 3 above)
Comment (filipecosta90) [reply src/model.ts]: solved in the lastest commit

File: src/model.ts (hunk 4 above)
Comment (filipecosta90) [reply src/model.ts]: solved in the lastest commit

**Comment by filipecosta90** [comment]
ready to be reviewed again @gkorland =)

File: src/model.ts [hunk 5]
```diff
@@ -11,14 +11,26 @@ export class Model {
...
+   * @param minbatchsize -  when provided with an minbatchsize that is greater than 0, the engine will postpone calls to AI.MODELRUN until the batch's size had reached minbatchsize
    */
-  constructor(backend: Backend, device: string, inputs: string[], outputs: string[], blob: Buffer | undefined) {
+  constructor(
+    backend: Backend,
+    device: string,
+    inputs: string[],
+    outputs: string[],
+    blob: Buffer | undefined,
+    batchsize?: number,
```
Comment (gkorland) [inline src/model.ts:23]: Default values:
batchsize = 0
minbatchsize = 0

File: src/model.ts [hunk 6]
```diff
@@ -11,14 +11,26 @@ export class Model {
...
+    minbatchsize?: number,
+  ) {
     this._backend = backend;
     this._device = device;
     this._inputs = inputs;
     this._outputs = outputs;
     this._blob = blob;
     this._tag = undefined;
+    this._batchsize = batchsize || 0;
```
Comment (gkorland) [inline src/model.ts:32]: maybe verify batchsize and minbatchsize are >=0?

File: src/model.ts [hunk 7]
```diff
@@ -86,14 +98,39 @@ export class Model {
     this._blob = value;
   }
 
+  private _batchsize: number;
+
+  get batchsize(): number {
+    return this._batchsize;
+  }
+
+  set batchsize(value: number) {
```
Comment (gkorland) [inline src/model.ts:107]: verify value >= 0?

File: src/model.ts (hunk 6 above)
Comment (filipecosta90) [reply src/model.ts:32]: adding a guard to it after initial check for undefined or assignement to zero.
```
this._batchsize = batchsize || 0;
    if (this._batchsize < 0) {
      this._batchsize = 0;
    }
```

**Comment by filipecosta90** [comment]
@gkorland I believe we're ready for review

**Pull request merged**